sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Monitor.OpenTelemetry.Exporter;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace Azure.Monitor.OpenTelemetry
+{
+    internal class AzureMonitorOpenTelemetryImplementations
+    {
+        internal static IServiceCollection AddAzureMonitorOpenTelemetryWithOptions(IServiceCollection services, AzureMonitorOpenTelemetryOptions? options)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            options ??= new AzureMonitorOpenTelemetryOptions();
+            var builder = services.AddOpenTelemetry();
+
+            if (options.EnableTraces)
+            {
+                builder.WithTracing(b => b
+                             .AddAspNetCoreInstrumentation()
+                             .AddAzureMonitorTraceExporter(o =>
+                             {
+                                 o.ConnectionString = options.ConnectionString;
+                                 o.DisableOfflineStorage = options.DisableOfflineStorage;
+                                 o.StorageDirectory = options.StorageDirectory;
```

Comment:
Minor, since the flag is using "OfflineStorage", maybe consider "OfflineStorageDirectory". Probably good to check how other language implementations are doing and align on the terminology.